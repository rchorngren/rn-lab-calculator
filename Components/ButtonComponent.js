import React, { useContext } from "react";
import { Pressable, StyleSheet } from "react-native";
import ButtonContext, { ButtonProvider } from "./ButtonContext";

// Using context rather than props
const ButtonComponent = ({ text, color, runFunction }) => {
  const buttonColor = color;
  const functionToRun = runFunction;

  const styles = StyleSheet.create({
    buttonStyle: {
      width: 50,
      height: 50,
      alignItems: "center",
      justifyContent: "center",
      borderRadius: 25,
      margin: 5,
      borderWidth: 2,
      borderColor: "black",
      backgroundColor: buttonColor,
    },
  });

  function ButtonText() {
    const textOnButton = useContext(ButtonContext);
    return <div>{textOnButton.buttonText}</div>;
  }

  return (
    <Pressable style={styles.buttonStyle} onPress={functionToRun}>
      <ButtonProvider value={{ buttonText: text }}>
        <ButtonText />
      </ButtonProvider>
    </Pressable>
  );
};

export default ButtonComponent;
