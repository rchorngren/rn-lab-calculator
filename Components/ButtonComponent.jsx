// component to display button with received text etc

import React from "react";
import { Text, Pressable, StyleSheet } from "react-native";

// Using context rather than props
const ButtonComponent = ({ text, color, runFunction }) => {
  const buttonText = text;
  const buttonColor = color;
  const functionToRun = runFunction;

  const styles = StyleSheet.create({
    buttonStyle: {
      width: 50,
      height: 50,
      borderRadius: 25,
      alignItems: "center",
      justifyContent: "center",
      margin: 5,
      backgroundColor: buttonColor,
    },
  });

  function getWeather() {
    // let long = 18.672295;
    // let lat = 59.1287;

    fetch(
      "https://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/18.672295/lat/59.128700/data.json"
    )
      .then((response) => response.json())
      .then((json) => {
        console.log(
          "Sikt: ",
          json.timeSeries[0].parameters[12].values,
          " kilometer"
        );
        console.log(
          "Vindstyrka: ",
          json.timeSeries[0].parameters[14].level,
          "m/s"
        );
      })
      .catch((error) => {
        console.error(error);
      });
  }

  function actionOnTouch() {
    getWeather();
  }

  return (
    <Pressable style={styles.buttonStyle} onPress={functionToRun}>
      <Text>{buttonText}</Text>
    </Pressable>
  );
};

export default ButtonComponent;
