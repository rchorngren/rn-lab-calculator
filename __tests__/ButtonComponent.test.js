import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ButtonComponent from '../Components/ButtonComponent';

it("Matches snapshoton DisplayComponent", () => {

    const tree = renderer.create(<ButtonComponent />).toJSON();
    expect(tree).toMatchSnapshot();

});
