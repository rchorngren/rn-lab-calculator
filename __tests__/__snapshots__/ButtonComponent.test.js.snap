// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshoton DisplayComponent 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": "black",
      "borderRadius": 25,
      "borderWidth": 2,
      "height": 50,
      "justifyContent": "center",
      "margin": 5,
      "width": 50,
    }
  }
>
  <Text />
</View>
`;
